### PR TITLE
'View in API' preference is not obvious

### LINKED ISSUE #6289: 'View in API' preference is not obvious
View in API is enabled by setting the developer setting in prefs - this is not obvious

Need to think about discoverability

### PR DESCRIPTION
**Pull request by aalves08**
Fixes #6289 

- splitting up user preferences belonging to `DEV` key and updating references throughout rancher 
- update user preferences page to individually control each one of them

**Before**
Screenshot 2022-09-23 at 12 29 23

**After**
Screenshot 2022-09-23 at 12 29 26

### CODE REVIEW

**Comment by aalves08** [comment]
As discussed with @kwwii , split of these particular user preferences should be under a section called `Advanced Features`, which can be left in the same position as is in terms of order and it should feature as first option `View in Api`.

Will need some help with the copy as it might not be appropriate. 🙏 

FYI @nwmac @richard-cox

**Comment by kwwii** [comment]
The copy could read:
- [ ] Enable "View in API"
- [ ] Show Namespaces managed by Rancher Manager (not intended for editing or deletion)
- [ ] Enable Dark/Light Theme keyboard shortcut toggle (shift+T)
- [ ] Enable Language Selection keyboard shortcut toggle (shift+L)
- [ ] Hide all Type Descriptions

File: shell/store/prefs.js [hunk 1]
```diff
@@ -91,7 +91,13 @@ export const TIME_FORMAT = create('time-format', 'h:mm:ss a', {
 });
 
 export const TIME_ZONE = create('time-zone', 'local');
+// DEV will be deprecated on v2.7.0, but is needed so that we can grab the value for the new settings that derived from it
+// such as: VIEW_IN_API, ALL_NAMESPACES, NO_LOCALE_SHORTCUT, THEME_SHORTCUT
 export const DEV = create('dev', false, { parseJSON });
+export const VIEW_IN_API = create('view-in-api', mapPref(DEV) || false, { parseJSON });
+export const ALL_NAMESPACES = create('all-namespaces', mapPref(DEV) || false, { parseJSON });
+export const NO_LOCALE_SHORTCUT = create('no-locale-shortcut', mapPref(DEV) || false, { parseJSON });
+export const THEME_SHORTCUT = create('theme-shortcut', mapPref(DEV) || false, { parseJSON });
```
Comment (jordojordo) [inline shell/store/prefs.js:100]: I tested coming from an instance with the original "Enabled Developer Tools & Features" or `DEV` checkbox both enabled and disabled.
When the checkbox was disabled originally, all of the new checkboxes would default to `true`. It seems the `mapPref(DEV) || false` is returning `true` regardless of the initial setting.

**Review by jordojordo — CHANGES_REQUESTED** [review]
Had a couple of comments/questions.

File: shell/pages/prefs.vue [hunk 2]
```diff
@@ -154,92 +158,92 @@ export default {
       </div>
     </div>
     <hr />
-    <h4 v-t="'prefs.theme.label'" />
+    <!-- Theme -->
+    <h4 v-t="'prefs.theme.label'" class="mt-20" />
```
[the remark this answers is not in the document]
Comment (aalves08) [reply shell/pages/prefs.vue:162]: updated spacing throughout all sections... Feels/looks better imo

File: shell/store/prefs.js (hunk 1 above)
Comment (aalves08) [reply shell/store/prefs.js:100]: Oh my, this was so wrong... What was I thinking? I didn't even test this properly... Sorry about that 🙏 

On my latest commit I've added the correct logic on the `prefs` store to inherit from `DEV`

File: shell/layouts/plain.vue [hunk 3]
```diff
@@ -62,7 +62,7 @@ export default {
         <ActionMenu />
         <PromptRemove />
         <AssignTo />
-        <button v-if="dev" v-shortkey.once="['shift','t']" class="hide" @shortkey="toggleTheme()" />
+        <button v-if="themeShortcut" v-shortkey.once="['shift','t']" class="hide" @shortkey="toggleTheme()" />
```
[the remark this answers is not in the document]
Comment (aalves08) [reply shell/layouts/plain.vue:65]: Tbh, I don't even know if this option to toggle locale even makes sense, but I've added to both layouts as suggested

**Comment by aalves08** [comment]
@jordojordo should be ready for review 🙏 

this commit https://github.com/rancher/dashboard/pull/6947/commits/859a582b3a5eb0895c0c631a7fc6cac3e35a16f8 has all of the necessary changes

File: shell/layouts/plain.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (jordojordo) [reply shell/layouts/plain.vue:71]: I don't feel like it's necessary, but if it's in one layout I would assume it should be in the others. Maybe we could bring this up to the team if we should keep this shortcut around.

File: shell/store/prefs.js (hunk 1 above)
Comment (jordojordo) [reply shell/store/prefs.js]: No worries! Looks like your changes fixed the problem :sunglasses:

**Review by jordojordo — APPROVED** [review]
lgtm. The prefs from a previous installation were inherited correctly, everything seems to work as expected.

**Pull request merged**